test/broken-pipe.test.js

```diff
@@ -52,6 +52,8 @@ t.test('let error pass through', ({ equal, plan }) => {
   process.nextTick(() => stream.emit('error', new Error('kaboom')))
 
   stream.on('error', (err) => {
-    equal(err.message, 'kaboom')
+    t.assert.strictEqual(err.message, 'kaboom')
   })
+
+  process.nextTick(end)
```

Comment:
Without `end`, node test exists before the end.

Reply:
Can we put it in `stream.on('end')`?

Reply:
Nope, I got this error.

<img width="992" alt="Screenshot 2025-05-06 at 19 00 56" src="https://github.com/user-attachments/assets/54be3d69-ff3d-465b-ab46-769fe00df396" />

Reply:
I suspect `nextTick` is going to be flaky. I think we should:

```js
test('thing', async t => {
	const { promise, resolve } = Promise.withResolvers()
	const stream = createStream()

	stream.on('end', () => resolve())

	await promise
})
```

Reply:
@jsumners, this is the only solution I found that works.
If I don't call the `stream.end()`, the promise is still pending, and the `close` event won't called.
Let me know if you have another solution in mind that I didn't catch.